detoks/src/main/java/nl/tudelft/trustchain/detoks/gossiper/TorrentGossiper.kt

```diff
@@ -32,27 +32,33 @@ class TorrentGossiper(
         val deToksCommunity = IPv8Android.getInstance().getOverlay<DeToksCommunity>()!!
         val randomPeers = pickRandomN(deToksCommunity.getPeers(), peers)
         val handlers = TorrentManager.getInstance(context).getListOfTorrents()
+        val profile = TorrentManager.getInstance(context).profile
         if (randomPeers.isEmpty() || handlers.isEmpty()) return
 
         val max = if (handlers.size < blocks) handlers.size else blocks
         val randomMagnets = handlers.random(max).map { it.makeMagnetUri() }
 
         if(randomMagnets.isNotEmpty())
             randomPeers.forEach {
+                val data = randomMagnets
+                    .filter { it2 -> profile.profiles.containsKey(MagnetLink.hashFromMagnet(it2)) }
+                    .map { it3 -> Pair(
+                        it3,
+                        profile.profiles[MagnetLink.hashFromMagnet(it3)]!!.hopCount + 1) }
```

Comment:
Don't all torrents have a profile? (Since if we add them from a gossip I think we also add a profile, and when booting we add a profile too I believe, though not sure)

Reply:
Removed the filter and added a new function increment hop count which checks if the profile exists